packages/collaboration-manager/src/CollaborationManager.ts

```diff
@@ -123,7 +123,13 @@ export class CollaborationManager {
     // Disable event handling
     this.#shouldHandleEvents = false;
 
-    this.applyOperation(operation);
+    if (operation instanceof BatchedOperation) {
+      operation.operations.forEach((op) => {
+        this.applyOperation(op);
+      });
+    } else {
+      this.applyOperation(operation);
+    }
```

Comment:
Better to move this condition into the `applyOperation` method since BatchedOperation is also an operation

Reply:
Ah, it's already there. So you can remove it here